api/oci/extensions/repositories/git/type.go

```diff
@@ -0,0 +1,155 @@
+	"ocm.software/ocm/api/tech/git"
+	"ocm.software/ocm/api/utils/accessio"
+	"ocm.software/ocm/api/utils/accessobj"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+const (
+	Type   = "GitRepository"
+	TypeV1 = Type + runtime.VersionSeparator + "v1"
+
+	ShortType   = "Git"
+	ShortTypeV1 = ShortType + runtime.VersionSeparator + "v1"
+)
+
+func init() {
+	cpi.RegisterRepositoryType(cpi.NewRepositoryType[*RepositorySpec](Type))
+	cpi.RegisterRepositoryType(cpi.NewRepositoryType[*RepositorySpec](TypeV1))
+	cpi.RegisterRepositoryType(cpi.NewRepositoryType[*RepositorySpec](ShortType))
+	cpi.RegisterRepositoryType(cpi.NewRepositoryType[*RepositorySpec](ShortTypeV1))
+}
+
+// Is checks the kind.
+func Is(spec cpi.RepositorySpec) bool {
+	return spec != nil && (spec.GetKind() == Type || spec.GetKind() == ShortType)
+}
+
+func IsKind(k string) bool {
+	return k == Type || k == ShortType
+}
+
+// RepositorySpec describes an CTF RepositoryImpl interface backed by a git RepositoryImpl.
+type RepositorySpec struct {
+	runtime.ObjectVersionedType `json:",inline"`
+	accessio.StandardOptions    `json:",inline"`
+
+	// URL is the git url of the RepositoryImpl to resolve artifacts.
+	URL string `json:"url"`
+
+	// Ref is the git ref of the RepositoryImpl to resolve artifacts.
+	// Examples include
+	// - heads/master
+	// - tags/v1.0.0
+	// - pull/123/head
+	// - remotes/origin/feature
+	// If empty, the default is set to HEAD.
+	Ref string `json:"ref,omitempty"`
+
+	// Author is the author of commits generated by the repository. If not set, is defaulted from environment and git
+	// configuration of the host system.
+	Author *Author `json:"author,omitempty"`
+
+	// AccessMode can be set to request readonly access or creation
+	AccessMode accessobj.AccessMode `json:"accessMode,omitempty"`
+}
+
+// Author describes the author of commits generated by the repository.
+type Author struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+}
+
+var _ cpi.RepositorySpec = (*RepositorySpec)(nil)
+
+var _ cpi.IntermediateRepositorySpecAspect = (*RepositorySpec)(nil)
+
+type Options struct {
+	*Author
+	accessio.Options
+}
+
+// NewRepositorySpecFromOptions creates a new RepositorySpec from options.
+func NewRepositorySpecFromOptions(mode accessobj.AccessMode, url string, opts Options) (*RepositorySpec, error) {
+	spec, err := NewRepositorySpec(mode, url, opts.Options)
+	if err != nil {
+		return nil, err
+	}
+	spec.Author = opts.Author
+	return spec, nil
+}
+
+// NewRepositorySpec creates a new RepositorySpec.
+func NewRepositorySpec(mode accessobj.AccessMode, url string, opts ...accessio.Option) (*RepositorySpec, error) {
+	o, err := accessio.AccessOptions(nil, opts...)
+	if err != nil {
+		return nil, err
+	}
+	o.Default()
+	return &RepositorySpec{
+		ObjectVersionedType: runtime.NewVersionedTypedObject(Type),
+		URL:                 url,
+		StandardOptions:     *o.(*accessio.StandardOptions),
+		AccessMode:          mode,
+	}, nil
+}
```

Comment:
I think I do, I was under the assumption that large repositories for git are probably not a good match anyways because we should be needing things like git lfs anyhow. (separate thing I did not implement now)

Given that assumption above, I was thinking memory might be fine. Want me to adjust still given that thought?